app/src/main/java/org/wikipedia/suggestededits/SuggestedEditsContributionsFragment.kt

```diff
@@ -0,0 +1,522 @@
+                count = SuggestedEditsUserStats.totalImageCaptionEdits
+            }
+            EDIT_TYPE_IMAGE_TAG -> {
+                view = imageTagsView
+                count = SuggestedEditsUserStats.totalImageTagEdits
+            }
+            else -> {
+                view = allTypesView
+                count = SuggestedEditsUserStats.totalEdits
+            }
+        }
+
+        contributionsCountText.text = getString(R.string.suggested_edits_spinner_item_text, count, resources.getQuantityString(R.plurals.suggested_edits_contribution, count))
+        for (filterView in filterViews) {
+            if (filterView == view) {
+                filterView.setEnabledStateUI()
+            } else {
+                filterView.setDisabledStateUI()
+            }
+        }
+    }
+
+    private fun loadDataBasedOnFilter() {
+        when (editFilterType) {
+            EDIT_TYPE_ARTICLE_DESCRIPTION -> {
+                articleAndImageContributions.addAll(articleContributions)
+            }
+            EDIT_TYPE_IMAGE_CAPTION -> {
+                for (imageContribution in imageContributions) {
+                    if (imageContribution.editType == EDIT_TYPE_IMAGE_CAPTION) {
+                        articleAndImageContributions.add(imageContribution)
+                    }
+                }
+            }
+            EDIT_TYPE_IMAGE_TAG -> {
+                for (imageContribution in imageContributions) {
+                    if (imageContribution.editType == EDIT_TYPE_IMAGE_TAG) {
+                        articleAndImageContributions.add(imageContribution)
+                    }
+                }
+            }
+            else -> {
+                articleAndImageContributions.addAll(articleContributions)
+                articleAndImageContributions.addAll(imageContributions)
+            }
+        }
+    }
+
+    private class HeaderViewHolder internal constructor(itemView: View) : DefaultViewHolder<View?>(itemView) {
+        var headerText: TextView = itemView.findViewById(R.id.section_header_text)
+        fun bindItem(date: String) {
+            headerText.text = date
+            headerText.setTextColor(ResourceUtil.getThemedColor(headerText.context, R.attr.colorAccent))
+        }
+    }
+
+    private class ContributionItemHolder internal constructor(itemView: SuggestedEditsContributionsItemView) : DefaultViewHolder<SuggestedEditsContributionsItemView?>(itemView) {
+        fun bindItem(contribution: Contribution) {
+            view.setTitle(contribution.description)
+            view.setDescription(contribution.title)
+            view.setImageUrl(contribution.imageUrl)
+            view.setIcon(contribution.editType)
+            view.setPageViewCountText(contribution.pageViews)
+            getPageView(view, contribution)
+        }
+
+        private fun getPageView(view: SuggestedEditsContributionsItemView, contribution: Contribution) {
+            val disposables = CompositeDisposable()
+
```

Comment:
Can we reuse the same `CompositeDisposable()` in the fragment instead of creating a new one?

Reply:
This is in a different class @cooltey ..: ContributionItemHolder... we will have to pass it to adapter and pass it to the viewholder... i thought this is simpler..